torchvision/datasets/imagenette.py

```diff
@@ -76,15 +76,13 @@ def __init__(
         }
         self._samples = make_dataset(self._image_root, self.wnid_to_idx, extensions=".jpeg")
 
-    def _check_exists(self) -> bool:
-        return self._size_root.exists()
+    def _check_integrity(self) -> bool:
+        return check_integrity(self._size_root.with_suffix(".tgz"), self._md5)
 
     def _download(self):
-        if self._check_exists():
```

Comment:
Thanks for the PR @imkasen .

@pmeier do you remember why we're erroring instead of just doing an early `return` like in other datasets e.g. `clevr` ?

https://github.com/pytorch/vision/blob/838ad6ccf6ef7032485671a54c1570876e366dc5/torchvision/datasets/clevr.py#L81-L83

Reply:
The general idea was to raise on `download=True` if we can't properly verify the integrity. For example, after a folder is extracted, there is no way for us to know if the images inside this folder have not been tempered with.

Contrast this with datasets like `MNIST` or `CIFAR`, where we have checksums for each file on record. Here we can properly check the integrity and thus there is no need to raise on `download=True`.

Now, both `CLEVRClassification` and `Imagenette` fall into the first category. I don't know why we didn't follow our own "rule" for the former, but it is likely too late to change it now. However, I would treat "relaxing" the rule for the latter with caution. I don't think anything has changed that warrants walking back on our previous decision. However, if you do, you probably want to do it for all datasets to keep things consistent.